Make cpan faster

spechenin opened the pull request:
This is just a technology preview, it is not meant to merge to master branch.
Bootstrap-perl runs cpan executable multiple times. And every time it is going to initialize CPAN.pm databases which is time-consuming. I offer to spawn cpan only once and use it as daemon.

spechenin committed:
command.log

spechenin committed:
caller_line()

spechenin committed:
add early exit point
if $no_run is declared
thus we can load bootstrap-perl and test subs while testing

renormalist commented:
Interesting. I like it but prefer to play around with it a bit, I try to have a look soon(tm).

Similar to the other PR, I can't merge without conflicts,  if you could rebase I will probably just go ahead and merge, just because I like it and would simply follow your approach.

Thank you very much, I really appreciate it!

spechenin committed:
print $USER
Print all messages to STDOUT through $USER.
This is helpful when testing to be silent, to collect and test all user output.

spechenin committed:
dry_run.pl

spechenin commented:
The PR is rebased on PR#7

spechenin committed:
cpan command

renormalist commented:
From a couple of sample runs it seems to work well. Thanks!
I just made a release to CPAN.

The only thing I changed is to call warn() instead of die() because my shell code in system is still assuming fails are ok and needs to better work with multiple consecutive runs.
